File: server/src/L2.Admin.Contracts/Models/CharacterSummary.cs
namespace L2.Admin.Contracts;

public sealed record CharacterSummary(
    Guid Id,
    string Name,
    Guid AccountId,
    string? Username,
    int RaceId,
    string? RaceName,
    int SexId,
    string? SexName,
    int BaseClassId,
    string? BaseClassName,
    int ActiveClassId,
    string? ActiveClassName,
    short Level,
    long Experience,
    DateTimeOffset CreatedAt,
    DateTimeOffset? DeleteAfter,
    string Status);

File: web/app/types/models/character-summary.ts
import type { CharacterStatus } from './character-status'

export interface CharacterSummary {
  id: string
  name: string
  accountId: string
  username: string | null
  raceId: number
  raceName: string | null
  sexId: number
  sexName: string | null
  baseClassId: number
  baseClassName: string | null
  activeClassId: number
  activeClassName: string | null
  level: number
  experience: number
  createdAt: string
  deleteAfter: string | null
  status: CharacterStatus
}
